server/src/main/java/com/cloud/network/vpn/Site2SiteVpnManagerImpl.java

```diff
@@ -347,7 +347,7 @@ public Site2SiteVpnConnection startVpnConnection(long id) throws ResourceUnavail
                 if (conn.isPassive()) {
                     conn.setState(State.Disconnected);
                 } else {
-                    conn.setState(State.Connected);
+                    conn.setState(State.Connecting);
```

Comment:
Is there any FSM or logic that ensures that connections don't get stuck in the new `Connecting` state?

Reply:
Nope but the additions to VirtualNetworkApplianceManagerImpl.java and Site2SiteVpnManagerImpl.java take care of it